crates/detect-targets/src/detect/macos.rs

```diff
@@ -1,15 +1,24 @@
 use crate::TARGET;
 use guess_host_triple::guess_host_triple;
 
-pub(super) const AARCH64: &str = "aarch64-apple-darwin";
-pub(super) const X86: &str = "x86_64-apple-darwin";
+const AARCH64: &str = "aarch64-apple-darwin";
+const X86: &str = "x86_64-apple-darwin";
+const UNIVERSAL: &str = "universal-apple-darwin";
+
+pub(super) fn detect_alternative_targets(target: &str) -> impl Iterator<Item = String> {
+    let is_aarch64 = target == AARCH64;
+    let is_x86 = target == X86;
+
+    is_aarch64
+        .then(|| X86.to_string())
+        .into_iter()
+        .chain((is_aarch64 || is_x86).then(|| UNIVERSAL.to_string()))
```

Comment:
This is very confusing, can you instead make it an if else or a match even if it repeats things a bit

Reply:
Yes, I've replaced that with a if-else block.